### ISSUE TITLE
main.py

### ISSUE THREAD

**Issue by JeonghyunKim0800** [JeonghyunKim0800/translate_implement#1]
import torch
from transformers import MarianMTModel, MarianTokenizer

# Load the pre-trained MarianMT model and tokenizer
def load_model_and_tokenizer(model_name):
    print(f"Loading model: {model_name}")
    tokenizer = MarianTokenizer.from_pretrained(model_name)
    model = MarianMTModel.from_pretrained(model_name)
    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
    model.to(device)
    model.eval()  # Ensure evaluation mode
    return model, tokenizer, device

# Translate a batch of sentences
def translate_sentences(sentences, model, tokenizer, device, num_beams=4):
    inputs = tokenizer(sentences, return_tensors="pt", padding=True, truncation=True)
    inputs = {key: value.to(device) for key, value in inputs.items()}
    with torch.no_grad():
        translated_tokens = model.generate(
            input_ids=inputs['input_ids'],
            attention_mask=inputs['attention_mask'],
            num_beams=num_beams,
            max_length=50,
            length_penalty=1.0
        )
    # Decode the translated sentences
    translated_sentences = [tokenizer.decode(t, skip_special_tokens=True) for t in translated_tokens]
    return translated_sentences

# Main function with CLI support
def main():
    print("Welcome to the Translation App!")
    model_name = 'Helsinki-NLP/opus-mt-en-ko'
    model, tokenizer, device = load_model_and_tokenizer(model_name)

    # User input: sentences to translate
    print("\nEnter sentences to translate (separate with a '|'). Example: 'I am happy.|How are you?'")
    input_text = input("Sentences: ").strip()
    sentences = [s.strip() for s in input_text.split('|') if s.strip()]
    
    if not sentences:
        print("No valid input provided. Exiting.")
        return

    # Translate sentences
    translated = translate_sentences(sentences, model, tokenizer, device)
    
    # Display the results
    print("\nTranslated Sentences:")
    for i, (src, tgt) in enumerate(zip(sentences, translated), 1):
        print(f"{i}. {src} -> {tgt}")

if __name__ == "__main__":
    main()
